plugins/@grouparoo/mailchimp/__tests__/export/export-profile.ts

```diff
@@ -420,17 +457,41 @@ describe("mailchimp/exportProfile", () => {
     expect(user["email_address"]).toBe(emails["otherEmail"]);
     expect(user["merge_fields"]["FNAME"]).toBe(alternativeName);
     expect(user["merge_fields"]["PHONE"]).toBe(newPhoneNumber);
-    expect(user["tags"].length).toEqual(3);
+    expect(user["tags"].length).toEqual(4);
     expect(user["tags"]).toEqual(
       expect.arrayContaining([
         expect.objectContaining({ name: listOne.toLowerCase() }),
         expect.objectContaining({ name: listTwo.toLowerCase() }),
         expect.objectContaining({ name: listThree.toLowerCase() }),
+        expect.objectContaining({ name: listFour.toLowerCase() }),
       ])
     );
+  });
 
-    const oldUser = await getUser(emails["alternativeEmail"]);
-    expect(oldUser).toBe(null);
+  test("can not delete a user if sync mode does not allow it but should remove the tags", async () => {
+    await expect(
+      runExport({
+        syncOperations: { create: true, update: true, delete: false },
+        oldProfileProperties: {
+          email_address: emails["otherEmail"],
+        },
+        newProfileProperties: {
+          email_address: emails["otherEmail"],
+        },
+        oldGroups: [listOne, listTwo, listThree], // missing listFour to make sure that we're only removing the tags we know about;
+        newGroups: [],
+        toDelete: true,
+      })
+    ).rejects.toThrow(/sync mode does not allow removing/);
+    user = await getUser(emails["otherEmail"]);
+    expect(user).not.toBe(null);
+    expect(user["email_address"]).toBe(emails["otherEmail"]);
+    expect(user["tags"].length).toEqual(1);
+    expect(user["tags"]).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({ name: listFour.toLowerCase() }),
```

Comment:
hmm, after the latest changes I would expect this test to fail, but it seems to be still passing- it should just throw and not remove anything anymore. we should probably first check out why this isn't failing, and then update it as needed to reflect the new intended behavior.